climada/entity/measures/base.py

```diff
@@ -206,7 +200,15 @@ def calc_impact(self, exposures, imp_fun_set, hazard, assign_centroids=True):
         """
 
         new_exp, new_impfs, new_haz = self.apply(exposures, imp_fun_set, hazard)
-        return self._calc_impact(new_exp, new_impfs, new_haz, assign_centroids)
+        # assign centroids if missing
+        if new_haz.centr_exp_col not in new_exp.gdf.columns:
+            LOGGER.warning(
+                "No assigned hazard centroids in exposure object after the "
+                "application of the measure. The centroids were assigned in impact calcualtion."
```

Comment:
I think we should clarify how to avoid the warning and what it implies:
```suggestion
                "No assigned hazard centroids in exposure object after the "
                "application of the measure. The centroids will be assigned during impact "
                "calculation. This is potentiall costly. To silence this warning, make sure "
                "that centroids are assigned to all exposures."
```